### ISSUE TITLE
Upstream compatibility

### ISSUE THREAD

**Issue by vit9696** [ReddestDream/OcQuirks#22]
Hi,

In addition to just rebuilding with the latest OpenCorePkg master you will also need to fix memory range alignment in DSC: https://github.com/acidanthera/OpenCorePkg/commit/f963012e9814beaa5931f1723b32ba1596870853 and update the configuration.

Also note, that manual MmioWhitelist configuration is required on all Haswell chipsets if you want to use DevirtualiseMmio. Therefore I strongly recommend merging https://github.com/ReddestDream/OcQuirks/pull/18.

**Comment by ReddestDream** [comment]
Thanks! I fixed the memory range in DSC and merged MmioWhitelist along with other changes.

I still need testers for the MmioWhitelist. None of my hardware needs it, and that has been holding back merging. Thanks!

**Comment by vit9696** [comment]
That's not all. You have an outdated macbuild.tool: https://github.com/ReddestDream/OcQuirks/blob/master/macbuild.tool

This tool does not make sure that mtoc is compatible, and may accidentally result in broken Windows setups. Please interface with ocbuild, and turn it into something similar to the new version:
https://github.com/acidanthera/AppleSupportPkg/blob/master/macbuild.tool

Also, you have pretty serious issues with memory management. You do not need and cannot free variables allocated on the stack, so the presence of code like `gBS->FreePool (&Status);` is trash, just need to remove it.

**Comment by ReddestDream** [comment]
Okay. I've updated the macbuild.tool script. I get the point about mtoc. I have had some people with troubling compiling because of mtoc, and this should resolve their issues. Thx!

With Download-Fritz's help, we've made corrections to the memory management and error handling that 1Revenger1 implemented along with the MmioWhitelist commit. I've definitely learned more today. :/

Any other advice, suggestions, pull requests, test results, etc you might have are much appreciated. 

I made OcQuirks originally based on code from al3x for my own use before I moved on to OC, then I published it hoping that Clover devs would see it and integrate it as an alternative to the now-outdated AptioMemoryFix and the other far worse memory fix drivers. Now I just maintain it for a few people who care. I admittedly don't use it myself outside of testing . . .
